build-system/common/utils.js

```diff
@@ -29,11 +29,13 @@ const ROOT_DIR = path.resolve(__dirname, '../../');
 /**
  * Cleans and builds binaries with --fortesting flag and
  * overriden config.
+ *
+ * @param {boolean} minified
  */
-function buildMinifiedRuntime() {
+function buildRuntime(minified = true) {
```

Comment:
I think we should get rid of the flag parsing logic in this function and replace it with direct calls to either [`dist()`](https://github.com/ampproject/amphtml/blob/ceef8eefddbf0fcb4b4acb9acd0da53bc9a3f95c/build-system/tasks/dist.js#L124-L154) or [`build()`](https://github.com/ampproject/amphtml/blob/ceef8eefddbf0fcb4b4acb9acd0da53bc9a3f95c/build-system/tasks/build.js#L73-L93). This will cause compilation to happen in-process (instead of having to `exec()` a new `gulp` task), and you will no longer have to parse `--core_runtime_only` or `--extensions` since the functions will do so themselves. Moreover, you'll benefit from being able to use other flags like `--extensions_from`, `--noextensions`, etc.

This requires two things to happen:

1. The easy part is calling `dist()` when `minified` is `true` and `build()` when `minified` is `false`.
2. The less easy part is to plumb `--fortesting` in to `dist.js`, `helpers.js`, etc.

I'm happy to help by doing the `fortesting` plumbing in a separate PR. Meanwhile, if you'd like to get unblocked on this PR, it should be sufficient to update it with just the flag change suggested in my other comment.

Reply:
After seeing your comment in chat, I initially tried it that way and solved (1) before running into (2) and then giving up. 

> Meanwhile, if you'd like to get unblocked on this PR

Yes please! I'll add a TODO.